plugins/woocommerce-blocks/assets/js/blocks/product-gallery/inner-blocks/product-gallery-thumbnails/block-settings/index.tsx

```diff
@@ -103,7 +103,7 @@ export const ProductGalleryThumbnailsBlockSettings = ( {
 					value={ context.thumbnailsNumberOfThumbnails }
 					onChange={ ( value: number ) =>
 						updateBlockAttributes( productGalleryClientId, {
-							thumbnailsNumberOfThumbnails: value,
+							thumbnailsNumberOfThumbnails: Math.round( value ),
```

Comment:
Do we want to do something similar on the frontend also to account for it there?

Reply:
I think this won't be necessary; the way we use this block attribute on the front end is by setting it as a limit of how many thumbnails we render, so we are not directly using the value from it.